### PR TITLE
Add test coverage for MainNavStore

### LINKED ISSUE #62: Add test coverage for MainNavStore
## 📝 Description

The `MainNavStore` in `conduit-frontend/frontend-decompose-logic/src/commonMain/kotlin/mikufan/cx/conduit/frontend/logic/component/main/MainNav.store.kt` currently lacks test coverage. This store handles critical navigation logic including user login state management and menu item switching, making it important to have comprehensive tests.

## 🎯 Objective

Create a test class `MainNavStoreTest` following the established testing patterns in the codebase.

### Test Class Structure
- Follow the naming convention: `MainNavStoreTest`
- Location: `conduit-frontend/frontend-decompose-logic/src/commonTest/kotlin/mikufan/cx/conduit/frontend/logic/component/main/MainNavStoreTest.kt`
- **MUST follow the testing guidance in [`common-doc/frontend/decompose-logic.md`](./common-doc/frontend/decompose-logic.md#mvikotlin-store-testing)**

#### 1. **Bootstrapper Logic Testing**
- Test automatic state switching based on `UserConfigKStore.userConfigFlow` changes
- Test transition from `UserConfigState.OnUrl` → `SwitchToNotLoggedIn` action
- Test transition from `UserConfigState.OnLogin` → `SwitchToLoggedIn` action  
- Test error handling for invalid `UserConfigState.Landing` state

#### 2. **State Transitions Testing**
- Test `SwitchToNotLoggedIn`: verify menu items change to not-logged-in state
- Test `SwitchToLoggedIn(username)`: verify menu items include Favourite with correct username
- Test `MenuIndexSwitching`: verify pageIndex updates correctly
- Test state consistency validation logic

#### 3. **Intent Handling Testing**
- Test `MainNavIntent.MenuIndexSwitching` with valid indices
- Test bounds validation for invalid menu indices
- Test no-op behavior when switching to current index

#### 4. **Edge Cases & Error Scenarios**
- Test state consistency validation (logged in without Favourite menu item)
- Test invalid index handling in menu switching
- Test username changes for already logged-in users

### Dependencies to Mock
- `UserConfigKStore` - mock the `userConfigFlow` using `every { ... } returns flowOf(...)`
- Use `dev.mokkery` library for mocking as established in other tests

### Reference Implementation
Look at existing test patterns in:
- `LandingPageStoreTest.kt` - for bootstrapper and flow testing patterns
- `AuthPageStoreTest.kt` - for simple state change testing  
- `MeStoreTest.kt` - for `UserConfigKStore` mocking patterns

### Testing Utilities Required
- `StandardTestDispatcher` for coroutine control
- `TestScope` for managing coroutine lifecycles
- `Channel` for state observation
- `LoggingStoreFactory` wrapper for debugging

## 🔗 Related

This issue was deferred from PR #61 where we decided to back off from creating the test case initially to focus on other priorities.

## ✅ Acceptance Criteria

- [ ] `MainNavStoreTest.kt` created following established patterns
- [ ] All public store behavior covered with tests
- [ ] Bootstrapper flow testing implemented
- [ ] State transition testing for all messages
- [ ] Intent handling testing with edge cases
- [ ] Error scenarios tested (bounds checking, state validation)
- [ ] Tests follow the patterns documented in `common-doc/frontend/decompose-logic.md`
- [ ] All tests pass consistently
- [ ] Proper resource cleanup implemented (`@AfterTest` with `store.dispose()`)

## 💡 Implementation Notes

- The store uses a bootstrapper that collects from `UserConfigKStore.userConfigFlow` - this will require careful mocking and flow testing
- Pay attention to the state consistency validation logic in the executor
- The store has complex logic for determining when to dispatch actions vs. no-op
- Menu index bounds validation should be thoroughly tested

### PR DESCRIPTION
**Pull request by CXwudi**
## Summary

- Added comprehensive test coverage for MainNavStore following established patterns
- Modified MainNavStoreFactory to accept autoInit parameter for proper bootstrapper testing
- Updated documentation with autoInit pattern guidance

## Key Features

**MainNavStoreTest.kt:**
- Bootstrapper logic testing (UserConfigState transitions)
- State transition testing for all messages
- Intent handling with bounds validation
- Error scenarios and edge cases
- Proper resource cleanup

**AutoInit Parameter:**
- Added to MainNavStoreFactory.createStore() method
- Defaults to true (production behavior)
- Can be set to false for testing to control bootstrapper timing

**Documentation:**
- Added bootstrapper testing section to common-doc/frontend/decompose-logic.md
- Explains autoInit pattern with examples

## Test Coverage

Covers all requirements from issue #62:
- ✅ Bootstrapper flow testing
- ✅ State transitions
- ✅ Intent handling
- ✅ Edge cases & error scenarios
- ✅ UserConfigKStore mocking
- ✅ Proper resource cleanup

Fixes #62

Generated with [Claude Code](https://claude.ai/code)

Co-authored-by: CXwudi <CXwudi@users.noreply.github.com>

### CODE REVIEW

**Comment by CXwudi** [comment]
@claude please fix:

1. import issue, see the CI failed log
2. I see that you are manually coding up the logic for calling `.init()` function on the newly created store based on `autoInit` parameter, but I think the `storeFactory.create()` function already has a parameter for autoInit, please check online for newest MVIKotlin doc
3. gemini report 3 suggestions, solve them all

**Comment by CXwudi** [comment]
@claude fix import problem: `MainNavStoreTest.kt:35:47 Unresolved reference 'returns'.` check the newest doc of mokkery (kotlin multiplatform mockking lib)

**Comment by CXwudi** [comment]
@claude Now we get:

```
MainNavStoreTest > testMenuIndexSwitchingNoOpWhenSameIndex FAILED
    kotlinx.coroutines.test.UncompletedCoroutinesError at TestBuilders.kt:353

MainNavStoreTest > testBootstrapperOnUrlToNotLoggedIn FAILED
    kotlinx.coroutines.test.UncompletedCoroutinesError at TestBuilders.kt:353

MainNavStoreTest > testMenuIndexSwitchingLoggedInState FAILED
    java.lang.IllegalArgumentException at MainNavStoreTest.kt:265

MainNavStoreTest > testSwitchToLoggedInStateTransition FAILED
    kotlinx.coroutines.test.UncompletedCoroutinesError at TestBuilders.kt:353

MainNavStoreTest > testBootstrapperLandingStateThrowsException FAILED
    java.lang.AssertionError at MainNavStoreTest.kt:104
```

for `UncompletedCoroutinesError`, I think the problem is that the store.dispose() in the `@AfterTest` should now move inside the `runTest` lambda function, before the `diposable.dispose()`, because `runTest()` expect no coroutine is still running, but the store is still running and waiting for `runTest()` to finish so it can close the store in `@AfterTest`

I think we need to update the documentation to avoid the usage of `@AfterTest`, instead each test case should have its own `afterTest()` function that are called inside the `runTest()` call

For other two exception

```
[testBootstrapperLandingStateThrowsException](https://scans.gradle.com/s/f6554or7vf4t6/tests/task/:frontend-decompose-logic:testDebugUnitTest/details/mikufan.cx.conduit.frontend.logic.component.main.MainNavStoreTest/testBootstrapperLandingStateThrowsException)
[:frontend-decompose-logic:testDebugUnitTest](https://scans.gradle.com/s/f6554or7vf4t6/tests/task/:frontend-decompose-logic:testDebugUnitTest)
[mikufan.cx.conduit.frontend.logic.component.main.MainNavStoreTest](https://scans.gradle.com/s/f6554or7vf4t6/tests/task/:frontend-decompose-logic:testDebugUnitTest/details/mikufan.cx.conduit.frontend.logic.component.main.MainNavStoreTest)
[View test history](https://scans.gradle.com/scans/tests)
Failed
java.lang.AssertionError: Expected an exception of class java.lang.IllegalStateException to be thrown, but was completed successfully.
```

```
[testMenuIndexSwitchingLoggedInState](https://scans.gradle.com/s/f6554or7vf4t6/tests/task/:frontend-decompose-logic:testDebugUnitTest/details/mikufan.cx.conduit.frontend.logic.component.main.MainNavStoreTest/testMenuIndexSwitchingLoggedInState)
[:frontend-decompose-logic:testDebugUnitTest](https://scans.gradle.com/s/f6554or7vf4t6/tests/task/:frontend-decompose-logic:testDebugUnitTest)
[mikufan.cx.conduit.frontend.logic.component.main.MainNavStoreTest](https://scans.gradle.com/s/f6554or7vf4t6/tests/task/:frontend-decompose-logic:testDebugUnitTest/details/mikufan.cx.conduit.frontend.logic.component.main.MainNavStoreTest)
[View test history](https://scans.gradle.com/scans/tests)
Failed
java.lang.IllegalArgumentException: Target index 2 is out of bounds. Valid range: 0-1, menuItems: [Feed, SignInUp]
```

File: conduit-frontend/frontend-decompose-logic/src/commonTest/kotlin/mikufan/cx/conduit/frontend/logic/component/main/MainNavStoreTest.kt [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  @Test
+  fun testBootstrapperLandingStateThrowsException() = runTest(testDispatcher) {
+    // When
+    mainNavStore.init()
+    
+    // Given/Then - bootstrapper should throw IllegalStateException for Landing state
+    // Exception is thrown in the bootstrapper coroutine, so we need to catch it there
+    try {
+      userConfigStateChannel.send(UserConfigState.Landing)
+      // Give some time for the bootstrapper to process the state and throw
```
Comment (CXwudi) [inline conduit-frontend/frontend-decompose-logic/src/commonTest/kotlin/mikufan/cx/conduit/frontend/logic/component/main/MainNavStoreTest.kt:110]: @claude use stateChannel.receive() with timeout instead of this `advanceUntilIdle()` as it is not available in web env

File: conduit-frontend/frontend-decompose-logic/src/commonTest/kotlin/mikufan/cx/conduit/frontend/logic/component/main/MainNavStoreTest.kt [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+      advanceUntilIdle()
```
Comment (CXwudi) [inline conduit-frontend/frontend-decompose-logic/src/commonTest/kotlin/mikufan/cx/conduit/frontend/logic/component/main/MainNavStoreTest.kt]: @claude use stateChannel.receive() with timeout instead of this `advanceUntilIdle()` as it is not available in web env

File: conduit-frontend/frontend-decompose-logic/src/commonTest/kotlin/mikufan/cx/conduit/frontend/logic/component/main/MainNavStoreTest.kt [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+import kotlin.test.assertFalse
+
+class MainNavStoreTest {
+  private val testDispatcher = StandardTestDispatcher()
+  private val userConfigStateChannel = Channel<UserConfigState>(Channel.UNLIMITED)
```
Comment (CXwudi) [inline conduit-frontend/frontend-decompose-logic/src/commonTest/kotlin/mikufan/cx/conduit/frontend/logic/component/main/MainNavStoreTest.kt:30]: let's also make it a `private lateinit var`, and we create the new channel everytime in `@BeforeTest`
@claude

Comment (CXwudi) [reply conduit-frontend/frontend-decompose-logic/src/commonTest/kotlin/mikufan/cx/conduit/frontend/logic/component/main/MainNavStoreTest.kt:30]: And in each test, make sure to close the `userConfigStateChannel`

**Review by CXwudi — COMMENTED** [review]
@claude fix comments
